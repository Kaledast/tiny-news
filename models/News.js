const mongoose = require('mongoose');

const newsSchema = new mongoose.Schema({
  title: {
    type: String
  },
  content: {
    type: String,
    default: 'No Content yet'
  }
});

module.exports = mongoose.model('News', newsSchema);
