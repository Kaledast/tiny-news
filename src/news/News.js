import React from "react";
import styled from "styled-components";
//mport PropTypes from "prop-types";
const NewsWrapperOuter = styled.section`
 
  display: flex;
  align-self: center;
  max-width: 500px;
  width: auto;
  padding: 13px;
  background: #060606;
  border-radius: 1%;
  border: 0.5px solid black;
`;
const NewsWrapper = styled.div`
  display: flex;
  flex-direction: column;
  background: #fff7d4;
  border: 13px #ef925e solid;
  background: white;
  border-radius: 1%;
  box-shadow: 0px 0px 15px 2px #060606;
  overflow-y: scroll;
`;

const ArticleTopic = styled.h2`
  text-align: center;
  color: #060606;
`;

const ContentSection = styled.section`
  margin: 4px;
`;

const DeleteButton = styled.button`
  background: #3c211a;
  margin: 10px;
  width: 50px;
  color: #ef925e;
`;

export default function News({ title, content }) {
  return (
    <NewsWrapperOuter>
      <NewsWrapper>
        <ArticleTopic>{title}</ArticleTopic>
        <ContentSection>{content}</ContentSection>
        <DeleteButton>delete</DeleteButton>
      </NewsWrapper>
    </NewsWrapperOuter>
  );
}
