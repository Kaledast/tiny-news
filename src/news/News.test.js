import React from 'react';
import News from './News.js';
import { ThemeProvider } from 'styled-components';
import renderer from 'react-test-renderer';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('News Component', () => {
  it('renders a News-Article with, timestamp, image, title and description', () => {
    const save = false;
    const onSave = jest.fn();
    const article = {
      title: 'textTitle',
      content: 'testContent',
      describtion: 'testDescription',
      url:
        'https://www.sciencealert.com/gorgeous-new-crater-spotted-on-the-red-planet-is-one-of-the-largest-we-ve-seen',
      publishedAt: '2019-06-18T06:43:51Z',
      urlToImage:
        'https://www.sciencealert.com/images/2019-06/processed/marsimpactcrater_1024.jpg'
    };

    const component = renderer.create(
      <ThemeProvider
        theme={{
          mode: 'normal'
        }}
      >
        <News
          key="article id or '404 not found'"
          onSave={onSave}
          save={save}
          article={article}
        />
      </ThemeProvider>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('has a link', () => {
    const article = {
      title: 'textTitle',
      content: 'testContent',
      describtion: 'testDescription',
      url:
        'https://www.sciencealert.com/gorgeous-new-crater-spotted-on-the-red-planet-is-one-of-the-largest-we-ve-seen',
      publishedAt: '2019-06-18T06:43:51Z',
      urlToImage:
        'https://www.sciencealert.com/images/2019-06/processed/marsimpactcrater_1024.jpg'
    };

    const callback = jest.fn(() => {
      console.log('saved called');
    });

    const news = shallow(
      <News
        theme={'normal'}
        key="article id or '404 not found'"
        save={false}
        article={article}
        onSave={callback}
      />
    );
    expect(
      news
        .dive()
        .dive()
        .contains('input')
    );

    const checkbox = news
      .dive()
      .dive()
      .find('input');

    checkbox.simulate('change');
    expect(callback).toHaveBeenCalled();
  });
});
//expect(callback).toHaveBeenCalled();
//const actualWindowLocation = global.window.location.pathname;
// expect(actualWindowLocation).not.toBe(global.window.location.pathname);
/*

    const location = JSON.stringify(window.location);
    delete window.location;

    Object.defineProperty(window, 'location', {
      value: JSON.parse(location)
    });

    Object.defineProperty(global.location, 'href', {
      value: 'http://localhost.com',
      configurable: true
    });

    const link = news.find(el => el.type === 'a');
    */
