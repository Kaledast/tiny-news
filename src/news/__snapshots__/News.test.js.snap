// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`News Component renders a News-Article with, timestamp, image, title and description 1`] = `
<div
  className="sc-bdVaJa gpiNoN"
>
  <div>
    <div
      className="sc-ifAKCX kUArjq"
    >
      <input
        checked={false}
        id="input"
        onChange={[Function]}
        type="checkbox"
      />
      <label
        htmlFor="input"
      />
    </div>
    <div
      id="date"
    >
      2019-Jun-18
    </div>
    <div
      className="sc-bwzfXH LXVVG"
    >
      textTitle
    </div>
    <img
      alt=""
      className="sc-EHOje liVHwF"
      onError={[Function]}
      src="https://www.sciencealert.com/images/2019-06/processed/marsimpactcrater_1024.jpg"
    />
    <section
      className="sc-htpNat duRMdf"
    >
      testContent
    </section>
    <a
      className="sc-bxivhb cGAVRy"
      href="https://www.sciencealert.com/gorgeous-new-crater-spotted-on-the-red-planet-is-one-of-the-largest-we-ve-seen"
    >
      <div />
      read more...
    </a>
  </div>
</div>
`;
